### PR TITLE
Implement resume upload & parsing logic

### PR DESCRIPTION
**Pull request by Dante-Capobianco**
In this PR, I completed the following

- Implement logic to both click on file upload or drag/drop a file into the marked input area, allowing both PDF & docx files
- Implement logic to call [Affinda API](https://docs.affinda.com/reference/createdocument) to parse resume data & return to be used to add to user profile data
- Implement logic to display loader while uploading & parsing of resume + determination of new skills/interests occurs
- Implement error handling logic to display meaningful messages across a variety of edge cases

Test cases

- User attempts to upload a document other than pdf/docx - error message provided indicating docx/pdf is needed
- User attempts to continue past resume page without uploading a valid document - error message provided indicating a document must be uploaded
- User attempts to submit a resume that has already been uploaded to the application - duplicate document error message provided:
Screenshot 2025-07-24 at 10 02 12 PM

- Something goes wrong with the resume upload/parsing or retrieving the complete list of skills/interests - detailed error messages provided

<div>
    <a href="https://www.loom.com/share/69a40f3dfd244500aef6115304dbd55f">
      <p>Frontend Walkthrough - Watch Video</p>
    </a>
    <a href="https://www.loom.com/share/69a40f3dfd244500aef6115304dbd55f">
      
    </a>
  </div>

### CODE REVIEW

File: frontend/src/components/CreateAccount.jsx [hunk 1]
```diff
@@ -54,8 +56,10 @@ const CreateAccount = () => {
           />
         );
       case 3:
+        return <CreateAccountStepThree />;
+      case 4:
         return (
-          <CreateAccountStepThree
+          <CreateAccountStepFour
```
Comment (oztheozoz) [inline frontend/src/components/CreateAccount.jsx:70]: Not anything you need to fix in this project but it helps to put semantic meaning into your naming for your components rather than just incrementing a number. An example is if step number 4 was the create account step where you add the skills, then we could name it CreateAccountStep4AddSkills. Code is read more often than written so it's best to optimize for the reader. Adding context will help the reader understand more easily the step they are reading.

**Comment by oztheozoz** [comment]
Nice job implementing this feature! Really cool!

File: frontend/src/components/CreateAccount.jsx (hunk 1 above)
Comment (Dante-Capobianco) [reply frontend/src/components/CreateAccount.jsx:70]: Very good point! Thanks for the insight. I can see how this makes it easier to navigate & understand how each section of my code correlates with the actual user-facing site. While this is something you mentioned I do not need to change, I can see the value behind it beyond just numbering so a fix is implemented in the next PR!

**Comment by Dante-Capobianco** [comment]
> Nice job implementing this feature! Really cool!

Thank you! I appreciate it & glad you were able to review the different components of this feature (e.g. difference between PDF vs docx import workflows)
